### ISSUE TITLE
Enhancement: Please add Waist Circumference as an optional variable

### ISSUE THREAD

**Issue by GerryT11** [PhilippKosarev/bmi#4]
Thank you for this great and beautiful BMI application. It would be great if you could add Waist Circumference as an optional variable to the computation. The reason for this is that there has been considerable criticism of the BMI, but the addition of the aist circumference variable does solve most problems of the BMI and gives a substantially better estimate of health risks.
In other words: the classic BMI (height and weight) + waist circumference is a substantially better measure.

How could it be implemented in the UI?
I think that it still good to have the normal BMI as default in the application, because it is so widely used. However, it would be great to have a button: "Add optionally Waist Circumference" which shows an additional field to allow entering the waist circumference. 
In the output, I suggest to simply add the info about the risk categorization (depending on the value of Waist Circumference) as suggested by the article [1]. It would be also nice to get the output of the computation of the BRI (body roundness index), as suggested by the article [2]. 

Please see:
[1] https://www.medicalnewstoday.com/articles/323543 This article gives a nice overview, why the addition of waist circumference makes sense. Furthermore, in the section "Weight, obesity, and health risks" the article suggests to give different risk categorizations based on the values of the waist circumference, but it still reports the usual BMI. Just the interpretation of the BMI changes based on the waist circumference.

[2] https://en.wikipedia.org/wiki/Body_roundness_index#Calculation This wikipedia article nicely shows the addition of waist circumference to the BMI. The section "Calculation" shows the computation of the so-called BRI (body roundness index) that uses the variables Height, Weight and Waist Circumference and also has nice figures on the measurement of waist and height. 

I hope you like this idea.

**Comment by philippkosarev** [comment]
Hey, thank you for creating the issue :)

It is a good idea to give more options to users, but I do have some questions.

1. Is there an advantage to using BRI over Waist to Height Ratio?
2. What values are considered healthy or unhealthy?

Also I've made a demo which you can download [here](https://drive.google.com/file/d/1SKvQ2aDeoaJfx8SGS7wKxUlUZSe_iZlO/view?usp=sharing), would be happy to hear your thoughts on it

**Comment by GerryT11** [comment]
Thank you for the quick reply and the demo. 
These are good questions. I talked to a friend and she told me the following to measure obesity and related health risks:

Relevant variables are: Weight, Height, Waist circumference, Hip circumference, Body fat, certain Blood values, and for the interpretation Age and Sex. In a tool like the BMI app, it is probably not necessary to include the variables Body fat and certain Blood values, because the first requires either good equipment to properly measure it or a doctor, the latter requires a doctor. This leaves us with Weight, Height, Waist circumference, Hip circumference, and potentially Age and Sex.

She suggests to start off with the BMI as base indicator and to use another indicator (or some other indicators) to further refine the interpretation of the BMI. This means the indicators cannot be seen as either or (like in your demo), but rather they add further information/interpretation to the BMI. In terms of the application, this could follow a left-to-right paradigm: Left the BMI and if the user enters the optional other variables, the other indicators are shows to the right together with further info on interpretation. Just an idea.

In terms of other indicator(s), she suggests not to use indicators like BRI or BSI, as they try to mingle all variables together in one computation. She rather suggests to use as a second indicator the Waist-hip Ratio [1] as an established indicator for estimating obesity, health, and as a risk factor for developing serious health conditions. It uses the variables Waist Circumference and Hip Circumference. The article [1] nicely shows how to interpret the ratio (the DGSP interpretation in the table seems to be the most refined one). The interpretation depends on the variable Sex of the person. 

According to her, the Waist-hip ratio is a better indicator in addition to the BMI than the Waist to Height [2] ratio. However, since the Waist to Height ratio uses with Waist Circumference one variable that would have been already entered to the BMI app by the user, it would be possible to additionally compute it. However, the interpretation of the Waist to Height ratio depends on the Age of the person. For the interpretation, it needs age-adjusted boundary values, which is nicely shown in [3] (in German, but the table is easy to understand). The critical value is > 0.5 for everyone under 40 years; the critical value is between 0.5 - 0.6 between the age of 40 and 50, while one adds 0.01 in every additional year within the range (e.g. if 42 years, the threshold is 0.52); and the critical value is > 0.5 for everyone above 50 years.

To sum up: It would be perfect if the BMI app allows to enter the variables Height and Weight (for the computation of the BMI), and optionally to add the variables Waist Circumference, Hip Circumference, Age and Sex (for the computation of the Waist-hip Ratio and the Waist to Height Ratio). 

The BMI app would then compute and display:
BMI: using Height, Weight
Waist-hip Ratio: using Waist circumference, Hip circumference, and for the interpretation Sex
Waist to Height Ratio: using Waist circumference, Height, and for the interpretation Age

What do you think?

[1] https://en.wikipedia.org/wiki/Waist%E2%80%93hip_ratio#Indicator_of_health
[2] https://de.wikipedia.org/wiki/Waist-to-Height_Ratio
[3] https://de.wikipedia.org/wiki/Waist-to-Height_Ratio#Berechnung

**Comment by philippkosarev** [comment]
Okay, so I made a bit of a mock-up 
![screenshot](https://github.com/user-attachments/assets/5d028fd6-a792-4d0b-8779-519e7cbdc006)
Due to the number of parameters the applet grows into a full-blown spreadsheet.
So I do have to wonder whether LibreOffice Calc would be a better solution for you, it could have as many inputs as one needs and retroactively compute any of the results, what do you think?

**Comment by GerryT11** [comment]
Thanks a lot for the mock-up. Indeed, it looks crowded if everything is visible right after the start-up of the BMI app. I think there are several ways to make it visually more pleasant and user friendly:

Generally, I think it would be good to visually separate the entry of Height and Weight from the other variables to make clear that the first two refer to the BMI ("Inputs BMI") and the next four refer to additional indicators ("Inputs Waist to Height Ratio and Waist-hip"). How could this be done visually in the UI? I don't know which possibilities/widgets are the in libadwaita, but these come to my mind:

My first idea is that the BMI app after start-up looks exactly the same as it is today, but with a button "Further investigation", "Investigate my BMI" or "Optional computations", "Optional indicators" or "Options". It the user clicks on the button, the four variables for the Inputs Waist to Height Ratio and Waist-hip ratio slide out at the bottom or to the right of the app to allow inputting the additional variables. This means, only if the user clicks on that button, all input fields are visible.

Another idea would be the way "Gnome Calculator" handles it: In the top bar, there is a button that can trigger "Basic" (which is default) and "Advanced" which shows many more calculation options. Calculator also keeps the already entered numbers if the user switches from Basic to Advanced so that the user doesn't have to enter anything again.  In the case of the BMI app, "Basic" could be e.g. named "BMI (basic)" and "BMI (advanced)" or "BMI investigator"

How do you like the ideas?










There could be

**Comment by philippkosarev** [comment]
I really like the idea of a drop down in the header bar, but unfortunately it seems like it just refuses to run a connected function when selecting a different mode. For now, we'll have to settle on a toggle button.
I've added BRI to the results which makes it more symmetrical compared to inputs, imo that helps a lot in terms of looking consistent and not overcrowded. Thresholds for healthy/unhealthy are also adjusted to be correct-ish (would highly appreciate if you could verify those).
The age/sex selection could be coming, if I thought of a good way to do that and got a bit more info on how that affects the thresholds (which could take some time).

Here's the new [demo](https://drive.google.com/file/d/1hMnWmW_y0rDV6HqRPJntJ4nHtKUo9nkS/view?usp=sharing)

**Comment by GerryT11** [comment]
The demo looks great. After start-up, the BMI app looks very clean and then the advanced mode arrives after clicking on the toggle. The advanced mode also looks nice and clean.

In terms of interpretation dependent on age and gender, my second comment in this thread shows the thresholds: Regarding the  Waist-hip Ratio the table in article [1] gives the details on interpretation between men and women. Regarding the Waist-hip ratio, the computation of the thresholds in dependency of age is in the comment, too. 
Regarding the BMI, the detailed table on the interpretation of the thresholds is here: https://en.wikipedia.org/wiki/Body_mass_index#Categories
Regarding the BRI, here is an interpretation: https://www.omnicalculator.com/health/bri#what-does-my-bri-result-mean (in the section "What does my BRI mean?). Another BRI calculator, which is frequently referenced is this one: https://webfce.com/bri-calculator/ Unfortunately, the WebFCE calculator shows other values as "healthy zone" compared to the other BRI calculator. The following article seems to confirm the thresholds in the Omnicalculator "Individuals with a BRI of less than 3.4 had a 25% increased mortality risk compared to the healthy range, while those with a BRI of 6.9 had a 49% increased risk." https://www.healthline.com/health-news/is-the-body-roundness-index-the-new-bmi-what-it-says-about-your-health Hence, regarding the thresholds of the BRI I am still slightly confused.

**Comment by GerryT11** [comment]
In the new demo, I assessed the thresholds (so far I verified only the BMI, the others will follow). Here are my findings on the computed BMI values:

BMI: The BMI thresholds do not seem to be entirely correct: The actual normal range (Healthy) starts at 18.5, which is the result of 60kg at 180cm. The BMI app, however already shows 59 kg as "healthy", although the BMI actually is 18.2, which is still considered underweight. On the other hand, the threshold to overweight is calculate correctly: while 80 kg is still normal (BMI 24.7), 81kg is overweight (BMI 25.0). This is absolutely correct.

Two suggestions: It would be also good if the BMI result output is given with one decimal, e.g. 18.5. Also, the BMI categories in the wikipedia page are the official ones and they have 7 categories (from Underweight (Severe thinness)  to Obese (Class III)). It would be great if the app could be as detailed as the seven categories.

**Comment by philippkosarev** [comment]
Thank you for pointing this out, I fixed the BMI issues and added gender and age selection.
Waist to height ratio now takes age into account, and waist to hip ratio now takes gender into account. The BMI thresholds for boys/girls aged 2 to 20 change depending on age and gender, but I haven't got a good way of calculating that yet.

On the WHO classification, it doesn't seem to be used by many countries, for example, in the UK the NHS uses the [Underweight/Healthy/Overweight/Obese](https://www.nhs.uk/health-assessment-tools/calculate-your-body-mass-index/calculate-bmi-for-adults) classification. I would like the thresholds to be relevant to most people (I am a bit biased toward the UK), and this classification seems to be the most common, at least of what I have seen so far.
Although I also think it would be great if there was an option to switch to the WHO classification; so if the feature set will exceed critical mass, I'll make a preferences dialogue to manage all the different options, including the different classifications.

The BRI does seem to be quite arbitrary, and not too useful. It's mostly there to round up the pack, for symmetry. Will probably replace it in the future with another metric if there is a more useful one.

Also now the advanced mode can be toggled using a drop-down menu in the header bar :)
Here's the newer [demo build](https://drive.google.com/file/d/1idAZKXkKyQiu0l7sIXwoRfKbWAWSvwn7/view?usp=sharing)

**Comment by GerryT11** [comment]
Thank you! The new demo build looks beautiful and is super functional. The header bar to switch between basic and advanced looks nice and is intuitive. We tried it out with our values and it now gives an excellent overview of the relevant indicators one needs to assess the health status of the body composition. 

Regarding children: The BMI calculator of a German health insurer (https://www.tk.de/service/app/2002866/bmirechner/bmirechner.app), for example, does not compute any value for everyone under 18, because the results are not validated for children. They refer to Percentile curves or growth curves that children doctors can apply.

Regarding UK and WHO, there is no actual difference. It is only that the WHO has some more fine-grained information in brackets. The exception is the middle category, where the UK uses a different wording ("a healthy weight"), which is absolutely fine:
underweight -> Underweight (Severe thinness)
underweight -> Underweight (Moderate thinness)
underweight -> Underweight (Mild thinness)
a healthy weight -> Normal range
overweight -> Overweight (Pre-obese)
obese -> Obese (Class I)
obese -> Obese (Class II)
obese -> Obese (Class III)
Therefore, I think that the easiest way is not to create extra preferences, but to use the UK naming for the middle category "a healthy weight" and the more fine-grained WHO naming for the other categories, because it is the same like in UK, but with more info in the brackets.

My friend very much liked the new version of the app, but she suggested to add some narrative "explanation of results" that explain what the displayed results actually mean for the person. There could be, for example at the bottom of the app a button \_explanation_/ which would pop up the narrative or flip out the narrative at the bottom. She also had some ideas how to do this. I will ask her again about that.

**Comment by GerryT11** [comment]
FYI: I've just noticed that the developer of another app used a legal disclaimer on the website (and maybe also in the app) about health information. He writes "The results of the calculations are approximate, are for informational purposes only and cannot be used for medical purposes." https://appcenter.elementary.io/com.github.alexkdeveloper.bmi/ I don't know if something like this is advised or necessary.

**Comment by philippkosarev** [comment]
Okay, I've limited the minimum age to 18, kept the UK-ish classification for the Basic mode and the WHO classification for the advanced mode.
I'm not a doctor myself so I can't explain the results in any meaningful way, but if you've got a solid idea, potentially approved by a medical professional, feel free to open an issue to discuss it.
As far as legal disclaimers go, I don't think that will be necessary, unless the WHO or NHS comes knocking on my door, at least for now.

At this stage I feel the features are well-implemented and don't seem to have any major bugs so I will be making a new release soon. You are welcome to close this issue if you think it's been resolved.

**Comment by GerryT11** [comment]
Thank you so much. The application looks excellent and the features are really well implemented.

**Comment by GerryT11** [comment]
Thank you, I've just installed v2.0 from Flathub and everything looks great so far. I close this issue as resolved. 
In case, I'll find any bug, I will report it.
